pkg/util/util.go

```diff
@@ -29,6 +30,10 @@ import (
 	"k8s.io/kubernetes/pkg/util/mount"
 )
 
+var (
+	volumeNameRE = regexp.MustCompile("^[a-zA-Z0-9_-]+$")
```

Comment:
Should you put a max length on this? The V2 schema won't work if the prefix is longer than X characters.

Reply:
the max length of volID will be calculated on 2 things. one is clusterID length and custom prefix length.
we need to decide what should be the max value for both

Reply:
Cluster name is known a priori based on the storage class.